File: 2482. Difference Between Ones and Zeros in Row and Column/Difference Between Ones and Zeros in Row and Column.py
class Solution:
    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        m = len(grid)
        n = len(grid[0])
        ones_row = [0] * m
        ones_column = [0] * n
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                ones_row[i] += grid[i][j]
                ones_column[j] += grid[i][j]
        result = []
        for i in range(len(grid)):
            result.append([])
            for j in range(len(grid[0])):
                result[i].append(2*(ones_row[i] + ones_column[j]) - m - n)
        return result

File: 2482. Difference Between Ones and Zeros in Row and Column/Difference Between Ones and Zeros in Row and Column.cpp
class Solution {
public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        vector<int> onesRow(m);
        vector<int> onesColumn(n);
        for (int i = 0; i < m; i++){
            for (int j = 0; j < n; j++){
                onesRow[i] += grid[i][j];
                onesColumn[j] += grid[i][j];
            }
        }

        vector<vector<int>> result(m, vector<int>(n));
        for (int i = 0; i < m; i++){
            for (int j = 0; j < n; j++){
                result[i][j] = 2*(onesRow[i] + onesColumn[j]) - m - n;
            }
        }
        return result;
    }
};
